### PR TITLE
feat:  Make name and description optional for AntaTests

### PR DESCRIPTION
**Pull request by gmuloc**
# Description

Make name and description optional class attributes which can be derived from the Class name and dostring respectively.

Fixes # (issue id)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/models.py [hunk 1]
```diff
@@ -326,10 +325,11 @@ def test(self) -> None:
         Python logger for this test instance.
     """
 
-    # Mandatory class attributes
-    # TODO: find a way to tell mypy these are mandatory for child classes - maybe Protocol
+    # Optional class attributes
     name: ClassVar[str]
     description: ClassVar[str]
+    # Mandatory class attributes
+    # TODO: find a way to tell mypy these are mandatory for child classes - maybe Protocol
```
Comment (carl-baillargeon) [inline anta/models.py:332]: Should we investigate in this PR? Not sure if:
```
class AntaTestProtocol(Protocol):
    categories: ClassVar[list[str]]
    commands: ClassVar[list[AntaTemplate | AntaCommand]]
```

Would work?

File: anta/reporter/csv_reporter.py [hunk 2]
```diff
@@ -58,8 +58,7 @@ def split_list_to_txt_list(cls, usr_list: list[str], delimiter: str = " - ") ->
 
     @classmethod
     def convert_to_list(cls, result: TestResult) -> list[str]:
-        """
-        Convert a TestResult into a list of string for creating file content.
+        """Convert a TestResult into a list of string for creating file content.
```
Comment (carl-baillargeon) [inline anta/reporter/csv_reporter.py:61]: I think we already talked about this but there's no way of enforcing this (not carriage return after the `"""`) in ruff right?

File: anta/models.py [hunk 3]
```diff
@@ -505,13 +505,22 @@ def save_commands_data(self, eos_data: list[dict[str, Any] | str]) -> None:
...
-        mandatory_attributes = ["name", "description", "categories", "commands"]
+        """Verify that the mandatory class attributes are defined and set name and description if not set."""
+        mandatory_attributes = ["categories", "commands"]
         for attr in mandatory_attributes:
             if not hasattr(cls, attr):
                 msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute {attr}"
                 raise NotImplementedError(msg)
 
+        if not hasattr(cls, "name"):
+            cls.name = cls.__name__
```
Comment (carl-baillargeon) [inline anta/models.py:516]:
```suggestion
        cls.name = getattr(cls, "name", cls.__name__)
```

File: anta/models.py [hunk 4]
```diff
@@ -505,13 +505,22 @@ def save_commands_data(self, eos_data: list[dict[str, Any] | str]) -> None:
...
 
     def __init_subclass__(cls) -> None:
-        """Verify that the mandatory class attributes are defined."""
-        mandatory_attributes = ["name", "description", "categories", "commands"]
+        """Verify that the mandatory class attributes are defined and set name and description if not set."""
+        mandatory_attributes = ["categories", "commands"]
         for attr in mandatory_attributes:
             if not hasattr(cls, attr):
                 msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute {attr}"
                 raise NotImplementedError(msg)
```
Comment (carl-baillargeon) [inline anta/models.py:513]: Old code but I think `AttributeError` is more appropriate here + collect all missing attributes and report them together provides a clearer error message and helps users fix all issues in one go.

```suggestion
    missing_attrs = [attr for attr in mandatory_attributes if not hasattr(cls, attr)]
    if missing_attrs:
        attrs_str = ', '.join(missing_attrs)
        msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute(s): {attrs_str}"
        raise AttributeError(msg)
```

File: anta/models.py [hunk 5]
```diff
@@ -505,13 +505,22 @@ def save_commands_data(self, eos_data: list[dict[str, Any] | str]) -> None:
...
 
+        if not hasattr(cls, "name"):
+            cls.name = cls.__name__
+        if not hasattr(cls, "description"):
+            if not cls.__doc__:
+                # No doctsring - raise
+                msg = f"Cannot set the description for class {cls.name}, either set it in the class definition or add a docstring to the class."
+                raise AttributeError(msg)
+            cls.description = cls.__doc__.split(sep="\n", maxsplit=1)[0]
+
```
Comment (carl-baillargeon) [inline anta/models.py:523]: When extracting the description from the docstring, should we also handle cases where the docstring might be empty or contain only whitespace?

Could do something like:
```
    if not hasattr(cls, "description"):
        doc = (cls.__doc__ or '').strip()
        if doc:
            cls.description = doc.split('\n', 1)[0]
        else:
            msg = (
                f"Cannot set the description for class {cls.name}. "
                f"Either set 'description' in the class definition or add a docstring to the class."
            )
            raise AttributeError(msg)
```

File: anta/reporter/csv_reporter.py (hunk 2 above)
Comment (gmuloc) [reply anta/reporter/csv_reporter.py:61]: it is already enforced in this PR cd pyproject D212 addition: https://docs.astral.sh/ruff/rules/multi-line-summary-first-line/

**Review by carl-baillargeon — APPROVED** [review]
LGTM

**Pull request merged**